lib/pure/ropes.nim

```diff
@@ -189,78 +188,62 @@ proc `&`*(a, b: Rope): Rope {.rtl, extern: "nroConcRopeRope".} =
 proc `&`*(a: Rope, b: string): Rope {.rtl, extern: "nroConcRopeStr".} =
   ## The concatenation operator for ropes.
   runnableExamples:
-    var
-      r1 = rope("Hello, ")
-      r2 = "Nim!"
-    
-    let r = r1 & r2
+    let r = rope("Hello, ") & "Nim!"
     doAssert $r == "Hello, Nim!"
+
   result = a & rope(b)
 
 proc `&`*(a: string, b: Rope): Rope {.rtl, extern: "nroConcStrRope".} =
   ## The concatenation operator for ropes.
   runnableExamples:
-    var
-      r1 = "Hello, "
-      r2 = rope("Nim!")
-    
-    let r = r1 & r2
+    let r = "Hello, " & rope("Nim!")
     doAssert $r == "Hello, Nim!"
+
   result = rope(a) & b
 
 proc `&`*(a: openArray[Rope]): Rope {.rtl, extern: "nroConcOpenArray".} =
-  ## The concatenation operator for an openarray of ropes.
+  ## The concatenation operator for an `openArray` of ropes.
   runnableExamples:
-    let s = @[rope("Hello, "), rope("Nim"), rope("!")]
-    let r = &s
+    let r = &[rope("Hello, "), rope("Nim"), rope("!")]
     doAssert $r == "Hello, Nim!"
+
   for item in a: result = result & item
 
 proc add*(a: var Rope, b: Rope) {.rtl, extern: "nro$1Rope".} =
   ## Adds `b` to the rope `a`.
   runnableExamples:
-    var
-      r1 = rope("Hello, ")
-      r2 = rope("Nim!")
+    var r = rope("Hello, ")
+    r.add(rope("Nim!"))
+    doAssert $r == "Hello, Nim!"
 
-    r1.add(r2)
-    doAssert $r1 == "Hello, Nim!"
   a = a & b
 
 proc add*(a: var Rope, b: string) {.rtl, extern: "nro$1Str".} =
   ## Adds `b` to the rope `a`.
   runnableExamples:
-    var
-      r1 = rope("Hello, ")
-      r2 = "Nim!"
+    var r = rope("Hello, ")
+    r.add("Nim!")
+    doAssert $r == "Hello, Nim!"
 
-    r1.add(r2)
-    doAssert $r1 == "Hello, Nim!"
   a = a & b
 
 proc `[]`*(r: Rope, i: int): char {.rtl, extern: "nroCharAt".} =
   ## Returns the character at position `i` in the rope `r`. This is quite
-  ## expensive! Worst-case: O(n). If `i >= r.len`, `\0` is returned.
+  ## expensive! Worst-case: O(n). If `i >= r.len or i < 0`, `\0` is returned.
```

Comment:
pre-existing but what's the rationale for not raising an `IndexDefect`?
seems like all this does is help hide bugs